pkg/api/persistentvolumeclaim/util.go

```diff
@@ -61,9 +61,7 @@ func dataSourceIsEnabled(pvcSpec *core.PersistentVolumeClaimSpec) bool {
 
 		}
 
-		if utilfeature.DefaultFeatureGate.Enabled(features.VolumeSnapshotDataSource) &&
-			pvcSpec.DataSource.Kind == volumeSnapshot &&
-			apiGroup == "snapshot.storage.k8s.io" {
```

Comment:
Should we keep the `VolumeSnapshotDataSource` checks in place until it is removed (i.e. in v1.21)?

Reply:
I followed the VolumePVCDataSource GA PR here:
https://github.com/kubernetes/kubernetes/pull/88686/files#diff-71e3b98f9a6bbf5b8421e26a7ba0c079f397cd8d49abacdad943c66a4f44f03dR641

When it was promoted to GA in 1.18 release, all the feature gate checks were removed and there was a note saying it will be removed in 1.20.

VolumeSnapshot feature is GA in 1.20, so I think this check should be removed.

Reply:
I followed a similar change here when VolumePVCDataSource is moved to GA:
https://github.com/kubernetes/kubernetes/pull/88686/files#diff-e2cf46a73af5aa5abfa443b871de0af32fb9ae9bbf349b33bad50c170f560f25R54

Reply:
I see there is a new `LockToDefault` bit that is set when flags go GA that prevents changing of flag. So it makes sense to remove the checks.